app/src/main/java/ch/epfl/sweng/studdybuddy/firebase/MetaGroup.java

```diff
@@ -128,17 +123,28 @@ public void accept(List<Pair> pairs) {
                     safeAddId(gId, pairs.get(i).getValue(), pairs.get(i).getKey(), uIds);
                 }
                 groupUsers.clear();
-                getUsersfromIds(uIds, groupUsers);
+                getUsersFromIdsAndConsume(uIds, groupUsers, consumer);
             }
         });
     }
 
+
+
+    public ValueEventListener getBuddiesNotInGroup(String gId, String uId,  List<User> participants, List<User> buddies ){
+        return getBuddiesAndConsume(uId, buddies, new LinkedList<>(), FirebaseConsumers.filterBuddies(gId, new ArrayList<>(), participants, this));
+    }
+
     public void pushGroup(Group g, String creatorId) {
         db.select(Messages.FirebaseNode.GROUPS).select(g.getGroupID().getId()).setVal(g);
         Pair pair = new Pair(creatorId,g.getGroupID().toString());
         db.select(Messages.FirebaseNode.USERGROUP).select(Helper.hashCode(pair)).setVal(pair);
     }
 
+    public void pushBuddies(Buddy buddy){
+        db.select(Messages.FirebaseNode.BUDDIES).select(buddy.hash()).setVal(buddy);
+    }
+
```

Comment:
(ignore) Note: we could have inserted MetaUser in the class hierarchy, extending Metabase and being parent to Metagroup

Reply:
yes it could be in this way. maybe will do in the next PR refactoring according to time constraints.